examples/sphere/shallow_water.jl

```diff
@@ -18,46 +19,154 @@ import OrdinaryDiffEq
 using OrdinaryDiffEq: ODEProblem, solve, SSPRK33
 
 using Logging
-usempi = get(ENV, "CLIMACORE_DISTRIBUTED", "") == "MPI"
 if usempi
     using ClimaComms
     using ClimaCommsMPI
-    const context = ClimaCommsMPI.MPICommsContext()
-    const pid, nprocs = ClimaComms.init(context)
-
-    # log output only from root process
-    logger_stream = ClimaComms.iamroot(context) ? stderr : devnull
-
-    prev_logger = global_logger(ConsoleLogger(logger_stream, Logging.Info))
-    atexit() do
-        global_logger(prev_logger)
-    end
 else
     import TerminalLoggers
-    global_logger(TerminalLoggers.TerminalLogger())
-    context = nothing
 end
+using ClimaCorePlots
+import Plots
+
+# Physical parameters needed
+struct Params{FT}
+    R::FT
+    Ω::FT
+    g::FT
+    D₄::FT
+end
+Params(::Type{FT}) where {FT} = Params{FT}(
+    FT(6.37122e6),
+    FT(7.292e-5),
+    FT(9.80616),
+    FT(1.0e16), # hyperdiffusion coefficient
+)
+abstract type AbstractTest end
 # This example solves the shallow-water equations on a cubed-sphere manifold.
 # This file contains five test cases:
+struct SteadyStateTest{FT, P} <: AbstractTest
+    u0::FT # 2 * pi * R / (12 * 86400)
+    h0::FT # 2.94e4 / g
+    α::FT
+    params::P
+end
+function SteadyStateTest(α::FT) where {FT}
+    params = Params(FT)
+    return SteadyStateTest{FT, typeof(params)}(
+        FT(2 * pi * params.R / (12 * 86400)),
+        FT(2.94e4 / params.g),
+        α,
+        params,
+    )
+end
```

Comment:
```suggestion
Base.@kwdef struct SteadyStateTest{FT, P}
    params::P
    u0::FT = FT(2 * pi * params.R / (12 * 86400))
    h0::FT = FT(2.94e4 / params.g)
    α::FT
end
SteadyStateTest{FT}(;params, kwargs...) where {FT} =
    SteadyStateTest{FT, typeof(params)}(;params, kwargs...)
```

This one was a bit more tricky, but it works if we move `params` to the first field, and make the others derived from it. Then we can call with `SteadyStateTest{FT}(;params=Params{FT}(;), α=some_value)`.

I think this similar pattern can be applied to the other (more complicated) subtypes of `AbstractTest`s.